lms/djangoapps/mobile_api/users/serializers.py

```diff
@@ -141,6 +186,123 @@ class Meta:
         lookup_field = 'username'
 
 
+class CourseEnrollmentSerializerModifiedForPrimary(CourseEnrollmentSerializer):
+    """
+    Serializes CourseEnrollment models for API v4.
+
+    Adds `course_status` field into serializer data.
+    """
+
+    course_status = serializers.SerializerMethodField()
+    course_progress = serializers.SerializerMethodField()
+    course_assignments = serializers.SerializerMethodField()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.course = modulestore().get_course(self.instance.course.id)
+
+    def get_course_status(self, model: CourseEnrollment) -> Optional[Dict[str, List[str]]]:
+        """
+        Gets course status for the given user's enrollments.
+        """
+        try:
+            block_id = str(get_key_to_last_completed_block(model.user, model.course.id))
+        except UnavailableCompletionData:
+            block_id = ''
+
+        if not block_id:
+            return None
+
+        path, unit_name = self._get_last_visited_block_path_and_unit_name(block_id)
+        if not path and unit_name:
+            return None
+
+        path_ids = [str(block.location) for block in path]
+
+        return {
+            'last_visited_module_id': path_ids[0],
+            'last_visited_module_path': path_ids,
+            'last_visited_block_id': block_id,
+            'last_visited_unit_display_name': unit_name,
+        }
+
+    @staticmethod
+    def _get_last_visited_block_path_and_unit_name(
+        block_id: str
+    ) -> Union[Tuple[None, None], Tuple[List['XBlock'], str]]:  # noqa: F821
```

Comment:
Hi, sure we will be able to update the format of linter directives, but what do you think about switching to other linters, for example [ruff](https://github.com/openedx/edx-platform/issues/32174) in the future? 
Maybe it's worth discussing smooth transition from pylint only supported comments to flake8 like # noqa, that are supported by most python linters nowadays, so it might ease the migration?

Reply:
That's an interesting point, and I think it could be worth discussing separately. Currently, though, we almost exclusively use pylint directives instead of noqa directives, so I'd like to stay consistent rather than starting a new pattern in this PR.

If we want to switch from pylint directives to noqa in the future, we could use a script to do it in bulk.

Reply:
Yes, I see, I'm going to change `# noqa` to `pylint: ignore=...`